src/server/pfs/s3/driver.go

```diff
@@ -72,28 +81,31 @@ func (d *MasterDriver) listBuckets(pc *client.APIClient, r *http.Request, bucket
 }
 
 func (d *MasterDriver) bucket(pc *client.APIClient, r *http.Request, name string) (*Bucket, error) {
+	var repo *pfs.Repo
 	branch := "master"
-	var repo string
-	parts := strings.SplitN(name, ".", 2)
-	if len(parts) == 2 {
+	parts := strings.SplitN(name, ".", 3)
```

Comment:
The resolution behavior is the thing that seems most important, since it impacts how users interact with it. I'd like to require as little change as possible, but anyone who used S3 to access the `stats` branch before will need to do something different regardless.